docs/cookbook/adding-ajax-load-more-button-into-pagination.md

```diff
@@ -0,0 +1,140 @@
+}
+```
+
+Then we will add `getPaginatedResult` method into `BrandFacade`.
+
+```php
+/**
+ * @param int $page
+ * @param int $limit
+ * @return \Shopsys\FrameworkBundle\Component\Paginator\PaginationResult
+ */
+public function getPaginatedResult(
+    $page,
+    $limit
+) {
+    $paginationResult = $this->brandRepository->getPaginationResult(
+        $page,
+        $limit
+    );
+    
+    return new PaginationResult(
+        $paginationResult->getPage(),
+        $paginationResult->getPageSize(),
+        $paginationResult->getTotalCount(),
+        $paginationResult->getResults()
+    );
+}
+```
+
+Next we will modify list twig template and create twig template for generation of paging controls and paginated items via ajax.  
+We will use `paginator.loadMoreButton(paginationResult, url('front_brand_list'), pageQueryParameter)` twig component that will asynchronously load next page when user clicks on its button. We will also define `pageQueryParameter` variable so it will have unique name and it will not interfere with other paging component on the same page.
+
+```twig
+{# ShopBundle/Resources/views/Front/Content/Brand/list.html.twig #}
+{% extends '@ShopsysShop/Front/Layout/layoutWithPanel.html.twig' %}
+
+{% block title %}
+    {{ 'Brand overview'|trans }}
+{% endblock %}
+
+{% block main_content %}
+    <div>
+        <h1>{{ 'Brand overview'|trans }}</h1>
+        {% include '@ShopsysShop/Front/Content/Brand/ajaxList.html.twig' with {paginationResult: paginationResult} %}
+    </div>
+{% endblock %}
+```
+
+```twig
+{# ShopBundle/Resources/views/Front/Content/Brand/ajaxList.html.twig #}
+{% import '@ShopsysShop/Front/Inline/Paginator/paginator.html.twig' as paginator %}
+{% set entityName = 'brands'|trans %}
+{% set pageQueryParameter = 'brandPage' %}
+
+<div>
+    <div class="js-list-with-paginator">
+        {{ paginator.paginatorNavigation(paginationResult, entityName, pageQueryParameter) }}
+        <ul class='list-images js-list'>
+            {% for brand in paginationResult.results %}
+                <li class="list-images__item">
+                    <a href="{{ url('front_brand_detail', { id: brand.id }) }}" class="list-images__item__block list-images__item__block--with-label">
+                        {{ image(brand, { alt: brand.name }) }}
+                        <span>{{ brand.name }}</span>
+                    </a>
+                </li>
+            {% endfor %}
+        </ul>
+        <div class="text-center margin-bottom-20">
+            {{ paginator.loadMoreButton(paginationResult, url('front_brand_list'), pageQueryParameter) }}
+        </div>
+        {{ paginator.paginatorNavigation(paginationResult, entityName, pageQueryParameter) }}
+    </div>
+</div>
+```
+
+After that we will modify `listAction` method in `BrandController` so `Brand` list page will be paginated and we will be able to integrate it into another list page that has other paginated items.  
```

Comment:
I am thinking - by modifying the `listAction` you will modify `brands-list` page as well - it will be paginated as well as a side effect :slightly_smiling_face: It did not get it from the first sight. So I would either mention it here or create a new action for that purpose